sale_commission/models/settlement.py

```diff
@@ -3,6 +3,13 @@
 from odoo import api, exceptions, fields, models, _
 
 
+def _execute_onchanges(records, field_name):
+    """Helper methods that executes all onchanges associated to a field."""
+    for onchange in records._onchange_methods.get(field_name, []):
+        for record in records:
+            onchange(record)
```

Comment:
At times, onchanges return a dict instead of modifying current recordset. That's a supported feature and thus this automation can easily break.

The safest options to do this are:

- Call the method directly: `record._onchange_whatever()`
- Use this technique, which imitates what the web client does:

  ```py
  record.update(record.onchange({}, field_name, {field_name: "1"}))
  ```

Reply:
@Yajo not in new API. Please put me an example.

Reply:
Support is implemented in [this method](https://github.com/odoo/odoo/find/11.0#L4875-L4879).

Reply:
I don't see the method

Reply:
Yikes, wrong link. See here: https://github.com/odoo/odoo/blob/fcb48b724152db876f6752560cce5a5d875e8f46/odoo/models.py#L4876-L4878

Reply:
OK, but then the code is a bit more complicated. I haven't seen anyway return values in Odoo code.